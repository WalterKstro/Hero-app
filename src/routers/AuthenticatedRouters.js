import { Switch, Route } from "react-router-dom"
import MarvelPage from "../componentes/marvel/MarvelPage";
import Navbar from "../componentes/ui/Navbar";
import DcPage from "../componentes/dc/DcPage";
import DetaillsScreen from "../componentes/detail/DetaillsScreen";
import Error404Page from "../componentes/error404/Error404Page";
import HomePage from "../componentes/home/HomePage";
import SearchHero from "../componentes/search/SearchHero";
const AuthenticatedRouters = () => {
    return (
        <>
            <Navbar/>
                <div className="container">
                    <Switch>
                        <Route exact path="/" component={HomePage}/>
                        <Route exact path="/dc" component={DcPage}/>
                        <Route exact path="/marvel" component={MarvelPage}/>
                        <Route exact path="/hero/:id_hero" component={DetaillsScreen}/>
                        <Route exact path="/search" component={SearchHero}/>
                        <Route component={Error404Page}/>
                    </Switch>
                </div>
        </>
    )
}

export default AuthenticatedRouters
