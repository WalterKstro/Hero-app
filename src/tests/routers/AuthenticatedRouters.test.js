import { mount, shallow } from "enzyme"
import { MemoryRouter } from "react-router"
import AuthContext from "../../auth/AuthContext"
import AuthenticatedRouters from "../../routers/AuthenticatedRouters"

describe('Testing in <AuthenticatedRouters/>', () => {


    test('should show correctly', () => {
        const user = {
            email: 'Walter Francisco',
            pass: '123'
        }
        const action = jest.fn()

        const rendered = mount(
            <MemoryRouter>
                <AuthContext.Provider value={{user, action}}>
                    <AuthenticatedRouters/>
                </AuthContext.Provider>
            </MemoryRouter>
        )
        expect(rendered.find('.text-white').text()).toBe(user.email)
        expect(rendered).toMatchSnapshot()
    })
    
})
