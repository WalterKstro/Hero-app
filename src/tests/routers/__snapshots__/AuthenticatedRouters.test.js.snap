// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AuthenticatedRouters/> should show correctly 1`] = `
Array [
  <nav
    className="sticky-top navbar navbar-expand-lg navbar-dark bg-dark py-0"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Brand"
          src="/assets/Brand.svg"
        />
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC Universe
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/marvel"
              onClick={[Function]}
            >
              Marvel Universe
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
        </ul>
      </div>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav ms-auto"
        >
          <li
            className="nav-item"
          >
            <button
              className="nav-link text-white fw-bold btn"
            >
              Walter Francisco
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Sign Out
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="home col d-flex flex-md-row flex-sm-column align-items-center justify-content-between"
      >
        <main>
          <h1>
            Frontend Developer
          </h1>
          <address>
            <strong>
              Walter Castro
            </strong>
            <strong
              className="d-block"
            >
              <a
                href="mailto:walterkstro@gmail.com"
              >
                walterkstro@gmail.com
              </a>
            </strong>
            <p>
              City Guatemala
            </p>
          </address>
        </main>
        <figure>
          <img
            alt="Code QR LinkedIn"
            src="/assets/linkedin.jpeg"
          />
        </figure>
      </div>
    </div>
  </div>,
]
`;
